projects/packages/scheduled-updates/tests/php/class-scheduled-updates-test.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * Test class for Scheduled_Updates.
+ *
+ * @package automattic/scheduled-updates
+ */
+
+/**
+ * Test class for Scheduled_Updates.
+ *
+ * @coversDefaultClass Scheduled_Updates
+ */
+class Scheduled_Updates_Test extends \WorDBless\BaseTestCase {
+
+	/**
+	 * Set up.
+	 *
+	 * @before
+	 */
+	protected function set_up() {
+		parent::set_up_wordbless();
+		\WorDBless\Users::init()->clear_all_users();
+
+		// Initialize the WordPress filesystem variable.
+		global $wp_filesystem;
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		WP_Filesystem();
+		$this->wp_filesystem = $wp_filesystem;
+
+		$this->plugin_dir = WP_PLUGIN_DIR;
+		$this->admin_id   = wp_insert_user(
+			array(
+				'user_login' => 'dumasdasdasmy_user',
+				'user_pass'  => 'dummy_pass',
+				'role'       => 'administrator',
+			)
+		);
+		wp_set_current_user( $this->admin_id );
+
+		// Ensure plugin directory exists.
+		$this->wp_filesystem->mkdir( $this->plugin_dir );
+
+		// init the hook
+		add_action( 'rest_api_init', array( 'Automattic\Jetpack\Scheduled_Updates', 'add_is_managed_extension_field' ) );
+
+		do_action( 'rest_api_init' );
+	}
+
+	/**
+	 * Clean up after test
+	 *
+	 * @after
+	 */
+	protected function tear_down() {
+		// Clean up the temporary plugin directory
+		$this->wp_filesystem->rmdir( $this->plugin_dir, true );
+
+		// Clean up the plugins cache created by get_plugins()
+		wp_cache_delete( 'plugins', 'plugins' );
+
+		parent::tear_down_wordbless();
+	}
+
+	/**
+	 * Simulate and test unmanaged plugins
+	 *
+	 * @covers ::add_is_managed_extension_field
+	 */
+	public function test_unmanaged_plugins() {
+		// direct
+		$plugin_name = 'direct-plugin';
+		$this->wp_filesystem->mkdir( "$this->plugin_dir/$plugin_name" );
+		$this->populate_file_with_plugin_header( "$this->plugin_dir/$plugin_name/$plugin_name.php", 'direct-plugin' );
+
+		// make sure the directory exists
+		$this->assertTrue( $this->wp_filesystem->is_dir( "$this->plugin_dir/direct-plugin" ) );
+
+		$request       = new WP_REST_Request( 'GET', '/wp/v2/plugins' );
+		$result        = rest_do_request( $request );
+		$plugin_result = $result->get_data()[0];
+
+		$this->assertSame( 'direct-plugin', $plugin_result['textdomain'] );
+		$this->assertSame( false, $plugin_result['is_managed'] );
+	}
+
+	/**
+	 * Simulate and test managed plugins
+	 *
+	 * @covers ::add_is_managed_extension_field
+	 */
+	public function test_managed_plugins() {
+		// managed, we simulate a symlink to a subdirectory inside a wp directory
+		$plugin_name = 'managed-plugin';
+		$target_dir  = "$this->plugin_dir/wordpress";
```

Comment:
I _think_ this should not work. I'll have to check, but my understanding is that they're symlinked from a `wordpress` directory in root.

Reply:
@vishnugopal Could you experiment with `realpath` instead of `readlink`? 

Try creating a `wordpress` folder within `/plugins` for example and symlink that. That should fail the test.
Only if the path begins with `/wordpress` we know it's a managed plugin.

Reply:
Thanks much for this, I didn't understand that `/wordpress` must in root earlier. See this for a patch: https://github.com/Automattic/jetpack/pull/36170